src/main/java/calculator/domain/InputText.java

```diff
@@ -0,0 +1,22 @@
+package calculator.domain;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class InputText {
+    private final String REGEX = "//(.)\n(.*)";
+    String inputText = null;
+
+    public InputText(String inputText) {
+        if (inputText == null || "".equals(inputText.trim())) {
+            inputText = "0";
+        }
+        this.inputText = inputText;
+    }
+
+
+    public String[] splitInputText() {
+        Matcher m = Pattern.compile(REGEX).matcher(inputText);
```

Comment:
해당 메서드가 호출될때마다 패턴을 컴파일하도록 되어있네요!
패턴 컴파일은 시간이 많이 걸리는 프로세스이며 자원을 많이 사용하는것으로 알고 있습니다.
 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다.
```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```
불필요한 객체 생성을 피하도록 캐싱하여 개선하면 좋을 것 같아요 :)

Reply:
아 그렇군요 그러면 패턴컴파일부분을 재사용하는 방식으로 신경써주면 좀 더 개선할 방향이 있군요
이런쪽으로는 생각도 못하고 있었는데 지적해주셔서 정말 감사합니다 앞으로는 이런 부분도 신경쓸 수 있도록 하겠습니다!!

```
private static final Pattern pattern = Pattern.compile(REGEX);

    public String[] splitInputText() {
        Matcher m = pattern.matcher(inputText);
```